src/components/shared/chart/Canvas.js

```diff
@@ -109,28 +109,41 @@ export default class ChartCanvas<HoveredItem> extends React.Component<
     const canvas = this._canvas;
     const { containerWidth, containerHeight, scaleCtxToCssPixels } = this.props;
     const { devicePixelRatio } = window;
-    const devicePixelWidth: DevicePixels = containerWidth * devicePixelRatio;
-    const devicePixelHeight: DevicePixels = containerHeight * devicePixelRatio;
+
     if (!canvas) {
       return;
     }
-    // Satisfy the null check for Flow.
-    const ctx = this._ctx || canvas.getContext('2d', { alpha: false });
-    if (!this._ctx) {
+
+    let ctx = this._ctx;
+    if (!ctx) {
+      ctx = canvas.getContext('2d', { alpha: false });
       this._ctx = ctx;
     }
+
+    // Round down to integer values.
```

Comment:
This current comment is not really useful as it just repeats what the code does. But maybe you can explain clearer that canvas.width/height are integer, and if we assign a float value they will be clamped down anyway. So by rounding down we ensure that the condition below actually works.